kinesis/consumer.py

```
@@ -675,6 +686,78 @@ def get_shard_status(self):
+        """
+        if self.skip_describe_stream:
+            raise ValueError("wait_ready() is not supported with skip_describe_stream=True")
+
+        # Ensure lifecycle is initialized (get_conn calls start() which describes the stream)
+        if self.shards is None:
+            try:
+                await asyncio.wait_for(self.get_conn(), timeout=timeout)
+            except asyncio.TimeoutError:
+                raise asyncio.TimeoutError(
+                    f"Consumer did not become ready within {timeout}s (timed out connecting to stream)"
+                )
```

Ruff (0.15.5)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)